### DISCUSSION TITLE
Fermata Auto stopped working

### DISCUSSION THREAD

**Discussion by mathaytus** [AndreyPavlenko/Fermata#122] [General]
mine worked until " i guess " this week. now app works on phone but NOT in car / android auto. keeps .
2022 Hyundai palisade.
worked great 1st week or so.
now NO app works correctly from AAAD . carstream, AA mirror, screen2auto etc.
tried rolling back android auto , then the car recognize it, makes me update it

**Comment by AndreyPavlenko** [comment]
Do you mean the above apps do not work anymore? What's your Android and AA version?

**Reply by Steveie36** [reply]
Hi my fermata auto doesn't work in my car anymore only on my phone is there a way you can reset fermata and start from scratch?

**Reply by Hunt4a** [reply]
It's not working for me too android 14

**Reply by Super-comix** [reply]
salut, quelqu'un a resolu le probleme? car idem pour moi, depuis la mise a jour de mon tel android fin decembre ca ne fonctionne plus :(

**Comment by Sofiane1985** [comment]
> Voulez-vous dire que les applications ci-dessus ne fonctionnent plus ? Quelle est votre version Android et AA ?

hello pavlenko, thank you for developing a wonderful app, we wish you will update fermata auto by adding the youtube video background function, this is the only feature that is missing to have an app better than cartube. cordially

**Comment by singhgurbir35** [comment]
Hi mine stopped working too on ford mach e. Phone app is working fine.
Indtalled just a couple of days ago it was working fine. I am using google pixel 6 device. Attaching the android auto release version screen shot.
![Screenshot_20211202-065537](https://user-images.githubusercontent.com/95428740/144447090-1429fad6-6ba6-47b9-9acf-a3257a960d95.png)

**Comment by mathaytus** [comment]
> Do you mean the above apps do not work anymore? What's your Android and AA version?

Android is 12  Fermata works on the phone still
crashes when connected to car.
Android auto is 7.0.614134 Tried rolling it back to 6 , car doesnt recognize it when I do.
AA is checked for unkown sources.
With it working in the car at 1st and now not IM GUESSING its something new like an update to Android 12 from 11 
or something.
NOTHING works. carstream, AA mirror etc

**Reply by emsss78** [reply]
Same problem here...after my Note 20 upgraded to Android 12...I even downgraded to a recent version 7.0.614103 and still the app doesn't launch in my car...i'll try downgrading to 6.9 and see if this works

**Reply by fcaronte** [reply]
> Same problem here...after my Note 20 upgraded to Android 12...I even downgraded to a recent version 7.0.614103 and still the app doesn't launch in my car...i'll try downgrading to 6.9 and see if this works

You need to install new 7.2.620124 and will work

**Reply by emsss78** [reply]
> > Same problem here...after my Note 20 upgraded to Android 12...I even downgraded to a recent version 7.0.614103 and still the app doesn't launch in my car...i'll try downgrading to 6.9 and see if this works
> 
> You need to install new 7.2.620124 and will work

I downloaded the version you mentioned and it worked :D Thank you so much !!!

**Comment by tarroka** [comment]
same problem for me. upgraded to android 12 3 days ago and everything was working fine after upgrade. now fermata and carstream showing in car but when you click on them they crash and nothing happen. They open in the phone normally but not the car.

**Comment by AndreyPavlenko** [comment]
Could someone capture logcat when it crashes?

**Comment by tarroka** [comment]
Attached the logcat[](url
[crash.txt](https://github.com/AndreyPavlenko/Fermata/files/7644080/crash.txt)
)

**Comment by AndreyPavlenko** [comment]
Thanks a lot!
May be someone else can capture logcat? I need to make sure that everyone are getting the same error.

**Comment by podimonanil** [comment]
Install Fermata Auto and Fermata control with King Installer.
It will work in car .

**Comment by mathaytus** [comment]
Not familiar with King Installer

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by podimonanil** [comment]
Direct download link is not available.
More details in
http://www.inceptive.ru
The app is available in Tera box.
Download instructions and installation instructions are available.

**Comment by podimonanil** [comment]
https://forum.xda-developers.com/t/fermata-media-player-audio-and-video-player-for-android-auto.4079519/

You can download Kinginstaller from the above link.
Uninstall both Fermata Auto and Fermata control.
Download new versions Fermata Auto 1.8.9 from Andrey Pavlenko s GitHub. And install both apps using King Installer.
Many many thanks to Andrey Pavlenko
For this wonderful app.

**Comment by mathaytus** [comment]
Is it going erase stuff off the phone.


________________________________

**Comment by podimonanil** [comment]
No, it just an app like AAAD.
No root required.
The advantage is that you can install an apk that is downloaded to your phone.
You can download Fermata Auto and Fermata control from official GitHub page and install both using KingInstaller 
It is safe Safe.......
Better than AAAD, because AAAD allows you to download only through it and only one app per month unless you get a pro version.

**Comment by mathaytus** [comment]
Ok. I paid, thats how I know none are working.  Carstream aa mirror etc

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by tarroka** [comment]
mine is installed with king installer already and I have the issue. I dont use AAAD.

**Comment by mathaytus** [comment]
I get error restoring when i try to download and open kinginstaller

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by mathaytus** [comment]
FYI. Uninstalled android auto.  Installed old version  ( 6.9.113543 ) then updated to one shown in pic. ( pic not showing )
After installing fermata again.
IT WORKS AGAIN

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by mathaytus** [comment]
Is working but just got a notification to update Android auto. It WILL be required to use soon .😡

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by tarroka** [comment]
what android version you have currently as the pic not showing in your reply?

**Comment by EdmiltonJr** [comment]
Device Samsung Galaxy S21+ with Android 11, Android Auto 7.0 and Fermata Auto and Control 1.8.9 not working. The Control's icon is show in the screen in the car, but the Fermata Auto dont displayed.

![Screenshot_20211204-075032_Android Auto.jpg](https://user-images.githubusercontent.com/6454218/144707182-88c19d8a-5dd1-497e-9cc6-9c6a46c51f27.jpg)

![Screenshot_20211204-082400_Android Auto.jpg](https://user-images.githubusercontent.com/6454218/144707631-e22410bd-388d-4a8b-8341-34b5ffb92072.jpg)

**Comment by Sofiane1985** [comment]
need to clear cache and delete data in androidauto, run androidauto in car first, it will work normally

**Comment by Carlos1988rm** [comment]
No, han capado todo hoy Fermata auto, screen2auto...  Ni caché ni nada

**Comment by Sofiane1985** [comment]
> Non, han capado todo hoy Fermata auto, screen2auto... Ni caché ni nada

I had the problem this morning, all the applications have disappeared (cartube screen2auto fermeta ...), there was just the icon cartube controller and fermeta controller that appears, I empty the cache delete the data then restart androidauto auto only in the car and surprise all apps appear again

**Comment by Carlos1988rm** [comment]
Y despues al rato desaparecen de nuevo

**Comment by Carlos1988rm** [comment]
Leee un poco del tema Google está de nuevo y desde hace un par de horas está prohibiendo las aplicaciones multimedia y de duplicación que provienen de AAAD.

 Estamos monitoreando el problema para evaluar la mejor solución.  Idealmente, las aplicaciones dentro de AAAD deben actualizarse para que se resuelva este problema.  AAAD estará listo una vez que la actualización de las aplicaciones esté disponible.

**Comment by mathaytus** [comment]
6.9.113543

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by EdmiltonJr** [comment]
I have received reports that on version 7.1 of Android Auto all is working.

**Comment by Muro0101** [comment]
> Device Samsung Galaxy S21+ with Android 11, Android Auto 7.0 and Fermata Auto and Control 1.8.9 not working. The Control's icon is show in the screen in the car, but the Fermata Auto dont displayed.
> 
> ![Screenshot_20211204-075032_Android Auto.jpg](https://user-images.githubusercontent.com/6454218/144707182-88c19d8a-5dd1-497e-9cc6-9c6a46c51f27.jpg)
> 
> ![Screenshot_20211204-082400_Android Auto.jpg](https://user-images.githubusercontent.com/6454218/144707631-e22410bd-388d-4a8b-8341-34b5ffb92072.jpg)

Also same problem here, fermato auto not showing in Android auto app fermato control is showing.

**Comment by DozeRmy** [comment]
Hi guys, not working on AA v7.1.6145xx

![Screenshot_20211205-003355_Android Auto](https://user-images.githubusercontent.com/95551414/144727812-078ae15e-d946-44ce-af99-30089808441c.jpg)

**Comment by Patriot26rus** [comment]
РЕБЯТ, нашли решение кто-нибудь с fermata avto???, пропадает в андроид авто, блочит google. Откатывал обновления андроид авто, потом ставил, сразу какое то время не пропадает, но все же блочит через время. Есть решения??????

**Comment by ThanosGr1980** [comment]
> Η συσκευή Samsung Galaxy S21+ με Android 11, Android Auto 7.0 και Fermata Auto και Control 1.8.9 δεν λειτουργεί. Το εικονίδιο του Control εμφανίζεται στην οθόνη του αυτοκινήτου, αλλά το Fermata Auto δεν εμφανίζεται.
> 
> ![Στιγμιότυπο οθόνης_20211204-075032_Android Auto.jpg](https://user-images.githubusercontent.com/6454218/144707182-88c19d8a-5dd1-497e-9cc6-9c6a46c51f27.jpg)
> 
> ![Screenshot_20211204-082400_Android Auto.jpg](https://user-images.githubusercontent.com/6454218/144707631-e22410bd-388d-4a8b-8341-34b5ffb92072.jpg)

I have the same problem since yesterday
![Screenshot_20211205-115228_Android Auto](https://user-images.githubusercontent.com/95566989/144742484-44100e37-4e90-49a0-ad56-46d58680a86f.jpg)
![Uploading Screenshot_20211205-115228_Android Auto.jpg…]()

**Comment by ThanosGr1980** [comment]
![Screenshot_20211205-115207_Android Auto](https://user-images.githubusercontent.com/95566989/144742629-c8d7a80c-15ab-414c-be54-330e9a35ce48.jpg)

**Comment by ThanosGr1980** [comment]
I have the same problem since yesterday.  has it been done ??

**Comment by ThanosGr1980** [comment]
adroid auto blames for the problem, does not let the apps (fermata auto, carstreem,  screen2auto...) put it in the car envy... 🙄 
when you uninstall android auto and re-update everything appears, but after a while they leave the screen again...

**Comment by DozeRmy** [comment]
> Hi guys, not working on AA v7.1.6145xx
> 
> ![Screenshot_20211205-003355_Android Auto](https://user-images.githubusercontent.com/95551414/144727812-078ae15e-d946-44ce-af99-30089808441c.jpg)

Fermata Control gone also...

**Comment by ThanosGr1980** [comment]
![Screenshot_20211205-223645_Android Auto](https://user-images.githubusercontent.com/95566989/144763200-1c82e300-f0b7-49d5-8423-ac504a9b3fed.jpg)
Κοίτα αυτό , IPcarTV έχει!, τα άλλα (fermata auto carstreem screen2autoo...) όχι. Wtf!!!

**Comment by Muro0101** [comment]
Same here, uninstall android auto then reupdate android auto.
It works for a while but when you restart your car it's GONE!!!

**Comment by denismagic** [comment]
> мой работал до "я думаю" на этой неделе. теперь приложение работает на телефоне, но НЕ в автомобиле / android auto. держит. Hyundai Palisade 2022 года. отлично поработал 1-ю неделю или около того. теперь НИКАКОЕ приложение не работает корректно с AAAD. carstream, зеркало AA, screen2auto и т. д. попробовал откатить android auto, потом машина его распознает, заставляет обновить



> Вы имеете в виду, что указанные выше приложения больше не работают? Какая у вас версия Android и AA?

У меня тоже не работает Fermata в автомобиле. Нет в списке приложений. В телефоне работает нормально. Как это исправить?

**Comment by Alfredo2190** [comment]
Buen día . Hoy a la fecha no está funcionando el FERMATA 1.8.9... Únicamente abre como reproductor de música.. No así de videos ni las otras aplicaciones de YouTube o iptv ,web, etc.

**Comment by mathaytus** [comment]
> Do you mean the above apps do not work anymore? What's your Android and AA version?

I think i answered on another thread , Android is 12 and AA was the newest .
not sure if it stopped when phone went from Android 11 to 12 or was a AA update.
BUT, When I rolled back to AA 6.9.613713 it works again until AA updates. I have auto update off but it still manages to chance.
once it does, i have to uninstall and reinstall the older version every time. 
IS there a way to stop that completely? from updating

**Comment by EdmiltonJr** [comment]
> > Do you mean the above apps do not work anymore? What's your Android and AA version?
> 
> I think i answered on another thread , Android is 12 and AA was the newest .
> not sure if it stopped when phone went from Android 11 to 12 or was a AA update.
> BUT, When I rolled back to AA 6.9.613713 it works again until AA updates. I have auto update off but it still manages to chance.
> once it does, i have to uninstall and reinstall the older version every time. 
> IS there a way to stop that completely? from updating 

No need to install. Only clear Android Auto cache and data, Fermata will work correctly for a while.

**Comment by mathaytus** [comment]
ok, cool.
so when stops just clear cache and data.👌

David


________________________________

**Comment by Patriot26rus** [comment]
> ладно, круто. поэтому, когда останавливается, просто очистите кэш и данные.👌 Дэвид
> […](#)
> ________________________________ От: EdmiltonJr ***@***.***> Отправлено: понедельник, 6 декабря, 2021 2:24 часов: AndreyPavlenko/фермата ***@***.***> Копия: mathaytus ***@***.***>; Автор ***@***.***> Тема: re: [AndreyPavlenko/фермата] фермата авто перестал работать (выпуск #92) Вы имеете в виду, что вышеуказанные приложения больше не работают? Какова ваша версия для Android и AA? Я думаю , что ответил в другой теме, Android 12, и AA был самым новым . не уверен, остановилось ли оно, когда телефон перешел с Android 11 на 12, или это было обновление AA. НО когда я откатился до AA 6.9.613713, он снова работает до обновления AA. У меня отключено автоматическое обновление, но оно все еще работает. как только это произойдет, мне придется каждый раз удалять и переустановить более старую версию. ЕСТЬ ли способ полностью остановить это? от обновления Нет необходимости в установке. Только очистите кэш и данные Android Auto, Fermata некоторое время будет работать корректно. — Вы получаете это, потому что вы являетесь автором темы. Ответьте на это электронное письмо напрямую, просмотрите его на GitHub[#92 (комментарий)](https://github.com/AndreyPavlenko/Fermata/issues/92#issuecomment-987109725)>, или unsubscribe><https://github.com/notifications/unsubscribe-auth/AWXYNZOUPP7SFGSFVHJUPWTUPUEXDANCNFSM5JF2UC7A>. Сортируйте уведомления на ходу с помощью GitHub Mobile для iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> или Android><https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

Это как? Откатить обновления АА и обратно установить? У меня так только работает, как только заглушил авто, в следующий запуск уже нет фермы.

**Comment by EdmiltonJr** [comment]
That's right.
For it to work momentarily, you don't need to reinstall any app, just clear the Android Auto cache and data.

However, we are having success performing a clean install (removing previously installed versions) through AAAD 1.4.3.

**Comment by mathaytus** [comment]
I have AAAD 1.4.3
Fermata auto ( 1.8.9 ) still stops showing up in Android auto once it updates. ( auto update off )
Im gonna try to keep AA cache clear and see if it keeps working.

David


________________________________

**Comment by Alejandro2454** [comment]
@EdmiltinJr
hello, can you explain in more detail how you are achieving success and
what do you mean by success?

**Comment by EdmiltonJr** [comment]
Removing any existing Fermata version and doing a fresh Fermata installation from AAAD 1.4.3 is working. Several reports of success performing this procedure since yesterday for Android versions 9, 10 and 11 and Android Auto 7.0.*.

**Comment by mathaytus** [comment]
I have Android 12

David


________________________________

**Comment by Alejandro2454** [comment]
@EdmiltonJr. Muchas gracias por la información,
Saludos,
Alejandro

**Comment by Patriot26rus** [comment]
> Удаление любой существующей версии Fermata и выполнение новой установки Fermata из AAAD 1.4.3 работает. Несколько сообщений об успешном выполнении этой процедуры со вчерашнего дня для версий Android 9, 10 и 11 и Android Auto 7.0.*.

А kinginstaller не подходит????

**Comment by ThanosGr1980** [comment]
> Η κατάργηση οποιασδήποτε υπάρχουσας έκδοσης Fermata και η πραγματοποίηση μιας νέας εγκατάστασης Fermata από το AAAD 1.4.3 λειτουργεί. Αρκετές αναφορές επιτυχούς εκτέλεσης αυτής της διαδικασίας από χθες για τις εκδόσεις Android 9, 10 και 11 και Android Auto 7.0.*.

thanks for this news !.  I hope it succeeds, I will try it... just wait a few days because I can not download yet in AAAD ...

**Comment by Gho5tr1d35** [comment]
Goodnight. I tried to install it with AAAD and it works 👍
only works with fermata auto. does not work with AA mirror carstream screen2auto

**Comment by EdmiltonJr** [comment]
> > Удаление любой существующей версии Fermata и выполнение новой установки Fermata из AAAD 1.4.3 работает. Несколько сообщений об успешном выполнении этой процедуры со вчерашнего дня для версий Android 9, 10 и 11 и Android Auto 7.0.*.
> 
> А kinginstaller не подходит????

In the my tests, not working.

**Comment by EdmiltonJr** [comment]
> > Η κατάργηση οποιασδήποτε υπάρχουσας έκδοσης Fermata και η πραγματοποίηση μιας νέας εγκατάστασης Fermata από το AAAD 1.4.3 λειτουργεί. Αρκετές αναφορές επιτυχούς εκτέλεσης αυτής της διαδικασίας από χθες για τις εκδόσεις Android 9, 10 και 11 και Android Auto 7.0.*.
> 
> thanks for this news !.  I hope it succeeds, I will try it... just wait a few days because I can not download yet in AAAD ...

Here all continue working right.
Remove any version installed and install from AAAD 1.4.3

**Comment by ThanosGr1980** [comment]
> > > Η κατάργηση οποιασδήποτε υπάρχουσας έκδοσης Fermata και η πραγματοποίηση μιας νέας εγκατάστασης Fermata από το AAAD 1.4.3 λειτουργεί. Αρκετές αναφορές επιτυχούς εκτέλεσης αυτής της διαδικασίας από χθες για τις εκδόσεις Android 9, 10 και 11 και Android Auto 7.0.*.
> > 
> > 
> > ευχαριστώ για αυτά τα νέα!. Ελπίζω να πετύχει, θα το δοκιμάσω... απλά περιμένετε λίγες μέρες γιατί δεν μπορώ να κατεβάσω ακόμα στο AAAD...
> 
> Εδώ όλοι συνεχίζουν να δουλεύουν σωστά. Καταργήστε οποιαδήποτε εγκατεστημένη έκδοση και εγκαταστήστε από το AAAD 1.4.3

thanks a lot !, I will try it in a few days ... we know what happens to the other applications, mirror, android2auto, carstreem...?, will they work??
Thanks from Greece!

**Comment by Patriot26rus** [comment]
И действительно работает, если установить через aaad 1.4.3, через kinginstaller не работает. Спасибо парни!!!!!!

**Comment by Gho5tr1d35** [comment]
another solution is to install AA with kinginstaller and install fermata auto fermata control AA mirror carstream and all the applications that disappear. install everything with kinginstaller. I already tested it and it works

**Comment by FastenM** [comment]
> another solution is to install AA with kinginstaller and install fermata auto fermata control AA mirror carstream and all the applications that disappear. install everything with kinginstaller. I already tested it and it works

I tried that on Android 12 galaxy s21 ultra on latest version of AA using kingInstaller it didn't help, also clearing the data of AA helps for a few minutes and then it stops working again!

**Comment by mathaytus** [comment]
{   I tried that on Android 12 galaxy s21 ultra on latest version of AA using kingInstaller it didn't help, also clearing the data of AA helps for a few minutes and then it stops working again! }

Same for me. Installer doesnt matter. Android 12 seems to be the problem.
You have to clear saved info in AA and basically re set it up every time.
going back to AA 6.9 helps but manages to eventually update to the newest version even with auto update off .


David


________________________________

**Comment by Alfredo2190** [comment]
Hoy a la fecha no funciona nada. Android 11 recién actualizado. Tampoco pude instalar kinginstaller.. Descargo la app y nada.. Pantalla negra y no hace nada... Descargo e instalo Fermat Auto y Fermat control y si funciona el control.. Solo reproduce música..
Ya borré cachet de todo.. Desinstalé, limpié y volví a instalar y nada.

**Comment by Alfredo2190** [comment]
Ayer postee que no funcionaba . Desinstalé Todo de forma manual. limpié cachet y volví instalar un fermata Auto desde la AAAD y funcionó !!!!.. Feliz. Graciassssss

**Comment by markinhosmrk** [comment]
Tried many things but still no luck... even recompiling with a different APP_ID parameter. Any other tips??
I'm on AA version 7.1. Thanks!

**Comment by ThanosGr1980** [comment]
Fermata auto 1.8.9 it's OK  for me. Bat the carstreem and ather... not. need to delete memory... 😤

Αποκτήστε το Outlook για Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by Gho5tr1d35** [comment]
nstall fermata auto with aaad 1.4.3 and creen2auto with aaease works AA 7.1 and android 10

**Comment by markinhosmrk** [comment]
Hi, I´ve been trying lots of things and was surprised when I connected to the car today and Fermata was available.
Not sure it will stay this way, but overall what I did was:

- There was a SQL script on XDA (could not find it right now, will try later and will put it here) that created a trigger to remove some lines from phenotype.db (AppValidation__blocked_apps_list, something like it). I executed it and still it wasn´t working.
- Messing around with the DB, on the Flags table there are some entries named AppValidation__allowed_package_list. It had two entries for Samsung messaging apps - I added the package name for Fermata, right before the :.
- Fermata was installed manually directly from GitHub and enabled with AA Tweaker. AA is the latest version.

Not sure exactly which one of the steps made it work!

**Comment by pako-tech** [comment]
Fermata Auto 1.8.10 
AA version 7.1.614574-release 
Android 12

Fermata installed by Kinginstaller

Fermata is working in Android

Fermata visible in AA but cannot be started (crashing)

**Comment by AndreyPavlenko** [comment]
> Hi, I´ve been trying lots of things and was surprised when I connected to the car today and Fermata was available. Not sure it will stay this way, but overall what I did was:
> 
> * There was a SQL script on XDA (could not find it right now, will try later and will put it here) that created a trigger to remove some lines from phenotype.db (AppValidation__blocked_apps_list, something like it). I executed it and still it wasn´t working.
> * Messing around with the DB, on the Flags table there are some entries named AppValidation__allowed_package_list. It had two entries for Samsung messaging apps - I added the package name for Fermata, right before the :.
> * Fermata was installed manually directly from GitHub and enabled with AA Tweaker. AA is the latest version.
> 
> Not sure exactly which one of the steps made it work!

I tried AA Tweaker and added the package name to AppValidation__allowed_package_list, but, unfortunately, it does not work for me. What is the script you mentioned?

**Comment by molinebis1** [comment]
I can do it work once if you clear data in Android Auto app.  The problem is that next time you connect it will crash again. Samsung s20 android 12.

**Comment by Alejandro2454** [comment]
@molinebis1 hola, consulta  @fcaronte que el te dará la solución. Saludos

**Comment by ugly69** [comment]
Hello!

I succeeded on my phone (Galaxy s21 5 g android 12):

Step 1:
Go to settings, apps, select android auto, tap a few times in the "version & permission info" field, turn on the development mode, go back to android auto home screen, enter developer mode settings (top right corner), set application mode to developer, and mark "unknown sources" checkbox.

Step 2:
Install AAAD, then from AAAD install Fermata auto.

**Comment by markinhosmrk** [comment]
Hi guys,

For those who have root, I think I found a way to avoid Android Auto removing apps:

1. Clear app data for AA and set up it on your car the way you like it
2. Use a terminal to change permission for AA shared_prefs directory under /data/data/com.google.android.projection.gearhead (I used root shell, not sure if it would work without it)

chmod -R 551 /data/data/com.google.android.projection.gearhead/shared_prefs

It has been working fine for a couple days but I guess you won't be able to change any setting.

**Comment by markinhosmrk** [comment]
Hi guys!

Doing more tests... it seemed that only working with shared_prefs eventually did not work. I made further changes and so far (~ 3 days and multiple force closes/reboots) everything still working fine.

1. I noticed some files being created under phenotype/shared dir. This one specifically, com.google.android.libraries.consentverifier#com.google.android.projection.gearhead.pb, is not present right after clear data but gets downloaded later on.

I emptied it with "echo "" > com.google.android.libraries.consentverifier#com.google.android.projection.gearhead.pb" - if it does not exist you can use touch to create an empty file - and also changed ownership of the entire shared/ dir to root and set permissions to 400 (only root user can read them). So far everything working.

I also saw that on the shared_prefs when I made them read only AA created new files. Maybe we can let them with default permissions and settings can be changed - I left them since I set it up the way I like it and don't think will need to change anything later.
My AA release is 7.5.121304, a beta from APKmirror.

Commands to test, right after clear data:

chown -R root:root
/data/data/com.google.android.projection.gearhead/files/phenotype/shared
touch /data/data/com.google.android.projection.gearhead/files/phenotype/shared/com.google.android.libraries.consentverifier#com.google.android.projection.gearhead.pb
chmod -R 400 /data/data/com.google.android.projection.gearhead/files/phenotype/shared/*

Let me know how it goes for you, I think I might be on the right track here!

Thanks,
Marcos

**Comment by markinhosmrk** [comment]
I can confirm that at least for me, changing permissions and ownership of /phenotype/shared/ directory did the trick and I no longer have problems with unauthorized apps disappearing.
I use basically Fermata and other settings with AA tweaker like the coolwalk interface, and they would revert/hide Fermata every time no matter what I tried.

Only downside is that it requires root.

**Reply by AndreyPavlenko** [reply]
Checked with an OnePlus device and this method does not work for me.

**Comment by AndreyPavlenko** [comment]
The version 1.9.0 should work for rooted devices with Magisk 24+.
To enable the app:

- Download the latest LSPosed zygisk module - https://github.com/LSPosed/LSPosed/releases.
- Launch Magisk Manager, go to Modules and install the module.
- Go to settings, enable Zygisk and reboot.
- Install the latest Fermata Auto - https://github.com/AndreyPavlenko/Fermata/releases.
- After the installation, LSPosed should detect a new module.
- Dial `*#*#5776733#*#*` to launch the LSPosed manager.
- Go to modules, enable Fermata Auto and restart.

**Reply by onemsiz** [reply]
Hello. First of all thank you for your labour.. But I did everything while you say, but still missing on AA.
First off all the code is wrong. True dial code is` *#*#5776733#*#*` for LSPosed. Fermata Auto enabled but still missing on Andoid Auto.
Restared phone, patched again with AIO Tweaker.. Nothing change.
phone: Oneplus 8 pro on android 13. AA version 8.4.624324

![WhatsApp Görsel 2022-12-05 saat 12 09 00](https://user-images.githubusercontent.com/119848360/205598178-6efcee4e-0c58-438c-a740-06824c7af220.jpg)

**Reply by AndreyPavlenko** [reply]
> patched again with AIO Tweaker

I thought, AIO Tweaker does not work for Android 12+. Anyway, you don't need it.
Try to clear AA cache and data.

**Reply by NicolaNauwynck** [reply]
> The version 1.9.0 should work for rooted devices with Magisk 24+. To enable the app:

Does that mean that non rooted devices on Android 13 will no longer be able to use Fermata in AA?

I'm using OnePlus 8t with android 11 and the only reason not to upgrade to android 13 for me is the lack of fermata support 🍾

**Reply by fcaronte** [reply]
> > The version 1.9.0 should work for rooted devices with Magisk 24+. To enable the app:
> 
> Does that mean that non rooted devices on Android 13 will no longer be able to use Fermata in AA?
> 
> I'm using OnePlus 8t with android 11 and the only reason not to upgrade to android 13 for me is the lack of fermata support 🍾 

Umh not exactly, seem that on OnePlus with working custom app keep them working, but if you uninstall and then try to install again won't work. Something from Oppo things are broken with install method of this app

**Reply by NicolaNauwynck** [reply]
> Umh not exactly, seem that on OnePlus with working custom app keep them working, but if you uninstall and then try to install again won't work. Something from Oppo things are broken with install method of this app

Thanks for the feedback, indeed already saw some comments about that, last month I did a fresh install of android 12 on my oneplus 8T but I then no longer got fermata working so I downgraded again to android 11. So if I would do an upgrade it should continue to work? Wonder if it will also work if I upgrade to android 13

**Reply by fcaronte** [reply]
> > Umh not exactly, seem that on OnePlus with working custom app keep them working, but if you uninstall and then try to install again won't work. Something from Oppo things are broken with install method of this app
> 
> Thanks for the feedback, indeed already saw some comments about that, last month I did a fresh install of android 12 on my oneplus 8T but I then no longer got fermata working so I downgraded again to android 11. So if I would do an upgrade it should continue to work? Wonder if it will also work if I upgrade to android 13
> 

Actually not, OnePlus with old android 11 is the only way to have working after update, maybe now fermata can work with magisk but I can't try, I have a mi10

**Reply by gaduaxe** [reply]
After installed LSPosed, it didnt work now, seem LSPosed's bug

**Reply by desaeverg** [reply]
Hi, i did the Steps but thé last one i don't understand whare do it have to dial thé numbers? 
![Screenshot_2023-01-19-22-09-21-98_785cfb1f0fb0c9a2030c9b38a1c3479a](https://user-images.githubusercontent.com/91951880/213560214-dd7afc0a-064b-4b01-a2e1-205ab154c7cc.jpg)

**Reply by AndreyPavlenko** [reply]
> whare do it have to dial thé numbers?
Phone dialer.

If you use Magisk 24+, you don't need Riru. You need LSPosed-zygisk module. Please, uninstall both modules and install LSPosed-zygisk only.

**Comment by AndreyPavlenko** [comment]
@onemsiz Thanks. Have you enabled zygisk? Please, make sure that you have installed LSPosed zygisk, but not riru version. Could you check the module log in LSPosed? Are there any errors?

**Reply by onemsiz** [reply]
Yes I've installed zygisk module. I didn't recieve any error. Here the log.zip Could you check it please..

![WhatsApp Görsel 2022-12-05 saat 16 48 05](https://user-images.githubusercontent.com/119848360/205653089-dda143ec-826c-47e4-b4a2-f134ecb82b82.jpg)
[log.zip](https://github.com/AndreyPavlenko/Fermata/files/10154719/log.zip)

**Reply by AndreyPavlenko** [reply]
Have you enabled zygisk in the magisk settings?

**Reply by onemsiz** [reply]
Yes enabled.

**Reply by AndreyPavlenko** [reply]
What's the magisk version?

**Reply by onemsiz** [reply]
The lastest version. v25.2..should i downgrade AA.

Edit: uninstalled aa and fermata. Downgraded to aa version 7.9 and I installed fermata again. Now I can see fermata auto on AA. I Will edit after use.

**Reply by AndreyPavlenko** [reply]
Try updating AA. The latest version works for me.

**Reply by fcaronte** [reply]
An Italian user, say me to try enable the "system framework" as show in the picture

**Reply by fcaronte** [reply]
![Screenshot_20221206-165011_OwlGram~2.png](https://user-images.githubusercontent.com/26366403/205960125-c8920054-5b53-483e-b3e0-1fea49dd5b5e.png)

**Reply by grantgrm** [reply]
Hello. I own a Oneplus 8T on Oxygenos 13, with latest AA version. I have tried everything, AAAD, AAStore, AAease, King Installer, but nothing seems to work I could only get Fermata Control to show on AA. ¿Any known workaround?

**Reply by NicolaNauwynck** [reply]
> Hello. I own a Oneplus 8T on Oxygenos 13, with latest AA version. I have tried everything, AAAD, AAStore, AAease, King Installer, but nothing seems to work I could only get Fermata Control to show on AA. ¿Any known workaround?

Have you tried the following: https://github.com/AndreyPavlenko/Fermata/discussions/122#discussioncomment-4307775

Can you let me know if it works for you?

I also have a OnePlus 8T but waited with upgrading from android 11 so far.

**Reply by grantgrm** [reply]
Hello. Thank you for your answer. My phone is not rooted so I can not use Magisk. ¿Is that the only known solution at the moment?

**Reply by NicolaNauwynck** [reply]
The only solution that I have found so far, personally I would also prefer a solution that does not involve rooting

**Comment by shimritmalka** [comment]
Oneplus 9rt oxygen 12. not working, the app are not in the android auto list. Someone help?

**Comment by Alfredo2190** [comment]
Hi... I have seen that many have problems with Fermata, I tell you
that I am still using the same Fermata auto version 1.8.10 on Android
11 and previously Android 10 and I continue to see everything in the
Android auto version 8.5.6. ..

**Comment by Alfredo2190** [comment]
in case the information helps

**Comment by ingenika** [comment]
https://androidautorepair.github.io/repair
